plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -143,6 +143,12 @@ func NodeRules() []rbac.PolicyRule {
 		pvcStatusPolicyRule := rbac.NewRule("get", "update", "patch").Groups(legacyGroup).Resources("persistentvolumeclaims/status").RuleOrDie()
 		nodePolicyRules = append(nodePolicyRules, pvcStatusPolicyRule)
 	}
+
+	// CSI
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIPersistentVolume) {
+		volAttachRule := rbac.NewRule("get").Groups(storageGroup).Resources("volumeattachments").RuleOrDie()
```

Comment:
will eventually (before exiting alpha) need a corresponding update in the node authorizer (similar to how the node authorizer only allows kubelets to access PV objects related to the node).

How does the kubelet know the name of the VolumeAttachment object to look up?

Reply:
csiMountMgr#SetUpAt appears to do a list:
```go
//TODO (vladimirvivien) consider using VolumesAttachments().Get() to retrieve
//the object directly. This requires the ability to reconstruct the ID using volumeName+nodeName (nodename may not be avilable)
attachList, err := c.k8s.StorageV1alpha1().VolumeAttachments().List(meta.ListOptions{})
```

listing all storage objects isn't something nodes can do today. to get a particular VolumeAttachment, it looks like both the kubelet and the node authorizer would need to re-derive the VolumeAttachment name, presumably using this approach:
```go
func getAttachmentName(volName, nodeName string) string {
	// TODO consider using a different prefix for attachment
	return fmt.Sprintf("pv-%s", hashAttachmentName(volName, nodeName))
}
```

is that name generation finalized and stable? once we have existent VolumeAttachment objects, we won't be able to easily change the naming scheme.

Reply:
Yep, have this on our list for beta. Kubelet generates the name from the node and volume names (see https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_attacher.go#L261).

Reply:
> csiMountMgr#SetUpAt appears to do a list:

Good catch!

> listing all storage objects isn't something nodes can do today. to get a particular VolumeAttachment, it looks like both the kubelet and the node authorizer would need to re-derive the VolumeAttachment name, presumably using this approach:

@liggitt Would it be ok to give node permissions to List all `VolumeAttachment` objects for this alpha? We'll take it as a requirement to generate the `VolumeAttachment` name and do only Gets for beta.

Reply:
> @liggitt Would it be ok to give node permissions to List all VolumeAttachment objects for this alpha?

If it's gated, off by default, and has a TODO in the policy file to switch to a get and enable the specific gets in the node authorizer, sure.

cc me on the issue tracking the authorizer bits, I can help with that as well

Reply:
@liggitt it will be feature-gated.  Will reach out for authorizer as well.